src/compas_slicer/post_processing/generate_raft.py

```diff
@@ -0,0 +1,178 @@
+    direction: str
+        horizontal: Create a raft in the horizontal direction
+        vertical: Create a raft in the vertical direction
+        diagonal: Create a raft int the diagonal direction
+    raft_layers: int
+        Number of raft layers to add. Defaults to 1
+    raft_layer_height: float
+        Layer height of the raft layers. Defaults to same value as used in the slicer.
+    """
+
+    # check if a raft_layer_height is specified, if not, use the slicer.layer_height value
+    if not raft_layer_height:
+        raft_layer_height = slicer.layer_height
+
+    logger.info("Generating raft")
+
+    # find if slicer has vertical or horizontal layers, and select which paths are to be offset.
+    if isinstance(slicer.layers[0], compas_slicer.geometry.VerticalLayer):  # Vertical layers
+        # then find all paths that lie on the print platform and make them brim.
+        paths_to_offset, _ = slicer.find_vertical_layers_with_first_path_on_base()
+
+    else:  # Horizontal layers
+        # then replace the first layer with a raft layer.
+        paths_to_offset = slicer.layers[0].paths
+
+    # get flat lists of points in bottom layer
+    all_pts = []
+    for path in paths_to_offset:
+        for pt in path.points:
+            all_pts.append(pt)
+
+    # get xy bounding box of bottom layer and create offset
+    bb_xy = bounding_box_xy(all_pts)
+    bb_xy_offset = offset_polygon(bb_xy, -raft_offset)
+    # bring points in the xy_offset to the correct height
+    for pt in bb_xy_offset:
+        pt[2] = slicer.layers[0].paths[0].points[0][2]
+
+    # calculate x range, y range, and number of steps
+    x_range = abs(bb_xy_offset[0][0] - bb_xy_offset[1][0])
+    y_range = abs(bb_xy_offset[0][1] - bb_xy_offset[3][1])
+
+    # get maximum values of the bounding box
+    bb_max_x_right = bb_xy_offset[1][0]
+    bb_max_y_top = bb_xy_offset[3][1]
+
+    # get point in bottom left corner as raft start point
+    raft_start_pt = Point(bb_xy_offset[0][0], bb_xy_offset[0][1], bb_xy_offset[0][2])
+
+    # create starting line for diagonal direction
+    if direction == "diagonal":
+        c = math.sqrt(2*(distance_between_paths**2))
+
+        pt1 = Point(raft_start_pt[0] + c, raft_start_pt[1], raft_start_pt[2])
+        pt2 = Point(pt1[0] - y_range, pt1[1] + y_range, pt1[2])
+        line = Line(pt1, pt2)
+
+    # move all points in the slicer up so that raft layers can be inserted
+    for i, layer in enumerate(slicer.layers):
+        for j, path in enumerate(layer.paths):
+            for k, pt in enumerate(path.points):
+                slicer.layers[i].paths[j].points[k] = Point(pt[0], pt[1], pt[2] + (raft_layers)*raft_layer_height)
+
+    for i in range(raft_layers):
+
+        iter = 0
+        raft_points = []
+
+        # create raft points depending on the chosen direction
+        while True and iter < 9999:  # to avoid infinite while loop in case something is not correct
```

Comment:
Here the `while True `doesn't do anything. The line can be replaced with `while iter < 9999`, for the same functionality.
(Sorry that I am being a bit pedantic!)